First code submission for MVC Pipeline

donker opened the pull request:
### Core Purpose
Introduces a new ASP.NET MVC-based rendering pipeline as an alternative to the traditional WebForms pipeline, allowing DNN to modernize its architecture while maintaining backward compatibility.
### Key Features
Dual Pipeline Architecture:
- Traditional WebForms pipeline continues through /default.aspx
- New MVC pipeline routes through /DesktopModules/MvcPipeline/Default/Page/{tabId}/{locale}

[Video showcase the general usage](https://www.youtube.com/watch?v=L7ZHTP8e_7w)

### Module Support:

[More info about MVC Module Control Implementation](https://github.com/dnnsoftware/Dnn.Platform/pull/6700)
[Razor+ Module Development Guide](https://github.com/dnnsoftware/Dnn.Platform/blob/feature/mvc-pipeline-old/DNN%20Platform/DotNetNuke.Web.MvcPipeline/razor-module-development.md)


- HTML module converted to MVC
- Sample modules demonstrating [Razor+](https://github.com/dnnsoftware/Dnn.Platform/tree/aa87d3fcd79268cd096f5674cc95d2bedccf1445/DNN%20Platform/Modules/Samples/Dnn.ContactList.Razor),  MVC and SPA patterns

[How to compile Samples](https://github.com/dnnsoftware/Dnn.Platform/tree/d39d9e4b2eec73c2d529ebb6c24b55880f6bda1d/DNN%20Platform/Modules/Samples)

### Configuration:
- Pipeline selection at portal and page level
- Settings UI integrated into Site and Page Settings
- URL rewriter modifications to route based on pipeline settings

[More info about MVC pipeline (routing / settings)](https://github.com/dnnsoftware/Dnn.Platform/pull/6721)

### New MVC Pipeline Projects

#### DotNetNuke.Web.MvcPipeline.csproj - New core MVC pipeline library
- Skins : Html Helpers (replacement for skin objects)
- Containers : Html Helpers (replacement for skin objects for containers)
- ModelFactories : construction of the Page Model (Model of the DefaultController)
- ModuleControl : Module Control infrastructure
- Utils : MvcViewEngine (Replacement for RazorEngine based on MVC 5 and can only be used inside HttpContext)
- UI/Utilities : MvcClientAPI (replacement for ClientAPI for mvc)

#### DDRMenu 
- Adapted to suport MVC Pipeline

### Core Platform Projects

#### DotNetNuke.Library
- Added Site and Page pipeline settings
- Added Module Control settings
- URL rewriter changes to route based on pipeline settings
- Update SPA (Html5) to use new CDF

#### DotNetNuke.Abstractions
- Added PagePipeline property to IPortalSettings

#### DotNetNuke.Web.Mvc
- Integration with new MVC pipeline
- Make actual DNN MVC modules run on the MVC Pipeline

#### HTML
- Add support for MVC Pipeline rendering
- Add MVC views and controllers
- This is implemented as true MVC (with form tags) using AJAX

#### Dnn.PersonaBar.UI
UI for pipeline settings in Site and Page Settings

#### DNNConnect.CKE
- HTML helper additions for CKEditor in MVC context

### Skins/Aperture
- Added mvc suport to the skin (Razor files)

[Skin conversion guide](https://github.com/dnnsoftware/Dnn.Platform/blob/0706d5901b8f16cdeb879a6af7ee7fea427e9f3f/DNN%20Platform/Skins/Aperture/SKIN_WEBFORMS_TO_MVC_GUIDE.md)

### Website
- MVC  Razor Layout
- MVC Default skin
- mvc.js minimal stuff needed for dnn to work
- Terms and privacy pages (razor files)
- DefaultController
- ModuleActions Controller

### AdminExperiance
- settings ui for new settings
- mvc controls for persona and edit bar

This PR is a first submission of the work of the MVC Pipeline team to bring the main project up to date with what has been developed in a forked repo.

### Documentation

[MVC Module Control Implementation](https://github.com/dnnsoftware/Dnn.Platform/pull/6700)
[MVC pipeline (routing / settings)](https://github.com/dnnsoftware/Dnn.Platform/pull/6721)
[Razor Module Development Guide](https://github.com/dnnsoftware/Dnn.Platform/blob/feature/mvc-pipeline-old/DNN%20Platform/DotNetNuke.Web.MvcPipeline/razor-module-development.md)
[How to compile Samples](https://github.com/dnnsoftware/Dnn.Platform/tree/d39d9e4b2eec73c2d529ebb6c24b55880f6bda1d/DNN%20Platform/Modules/Samples)

[DNN MVC Pipeline - Developer User Guide](https://github.com/dnnsoftware/Dnn.Platform/blob/feature/mvc-pipeline-old/DNN%20Platform/DotNetNuke.Web.MvcPipeline/DNN_MVC_PIPELINE_USER_GUIDE.md)

---
# Review Tasklist

- [ ] Walkthrough: Ask the author for a live, screen-share session for explaining the high-level intent, architectural changes, and where the high-risk logic lives.

## 1) Prerequirement, Build and packaging (9 folders, A: 2, M: 9)

- [ ] DNN Platform/Library/Properties
- [ ] DNN Platform/Library/UI/UserControls
- [ ] DNN Platform/Library/Services/Tokens
- [ ] DNN Platform/Library/Services/Tokens/PropertyAccess
- [ ] DNN Platform/Library/UI/Modules
- [ ] DNN Platform/DotNetNuke.ContentSecurityPolicy
- [ ] DNN Platform/DotNetNuke.ContentSecurityPolicy/Properties
- [ ] Build/Tasks
- [ ] DNN Platform/Components/Microsoft.AspNetMvc

## 2) Settings (portal, page, modules) (19 folders, A: 3, M: 29)

- [ ] DNN Platform/DotNetNuke.Abstractions/Portals
- [ ] DNN Platform/Library/Entities/Modules
- [ ] DNN Platform/Library/Entities/Portals
- [ ] DNN Platform/Library/Entities/Tabs
- [ ] DNN Platform/Library/Data
- [ ] DNN Platform/Library/Services/Upgrade
- [ ] DNN Platform/Library/Services/Installer/Writers
- [ ] Dnn.AdminExperience/ClientSide/Pages.Web/src/components/More
- [ ] Dnn.AdminExperience/ClientSide/Pages.Web/src/services
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/DTO/SiteSettings
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Pages/App_LocalResources
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.SiteSettings/App_LocalResources
- [ ] Dnn.AdminExperience/ClientSide/Extensions.Web/src/components/EditExtension/CustomSettings/Module/ModuleDefinitions/ModuleDefinitionRow/Controls
- [ ] Dnn.AdminExperience/ClientSide/SiteSettings.Web/src/components/moreSettings
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Extensions/Dto/Editors
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/DTO
- [ ] Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.Extensions/App_LocalResources
- [ ] DNN Platform/Website/Providers/DataProviders/SqlDataProvider

## 3) Rewriter (1 folders, A: 0, M: 1)

- [ ] DNN Platform/Library/Entities/Urls

## 4) MvcPipeline (19 folders, A: 108, M: 0)

- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Containers
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Exceptions
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Extensions
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Framework
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Framework/JavascriptLibraries
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/ModelFactories
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Models
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/ModuleControl
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/ModuleControl/Page
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/ModuleControl/Razor
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/ModuleControl/WebForms
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Modules
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Pages
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Routing
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Skins
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/UI/Utilities
- [ ] DNN Platform/DotNetNuke.Web.MvcPipeline/Utils

# 5) Website & Admin (17 folders, A: 23, M: 5)
- [ ] DNN Platform/Website
- [ ] DNN Platform/Website/Controllers
- [ ] DNN Platform/Website/Models
- [ ] DNN Platform/Website/Resources/Shared/scripts/jquery
- [ ] DNN Platform/Website/Views
- [ ] DNN Platform/Website/Views/Default
- [ ] DNN Platform/Website/Views/PersonaBarContainer
- [ ] DNN Platform/Website/admin/Portal/Controls
- [ ] DNN Platform/Website/admin/Portal/Views
- [ ] DNN Platform/Website/js
- [ ] Dnn.AdminExperience/EditBar/Dnn.EditBar.UI
- [ ] Dnn.AdminExperience/EditBar/Dnn.EditBar.UI/Mvc
- [ ] Dnn.AdminExperience/Library/Dnn.PersonaBar.UI
- [ ] Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/Mvc

## 6) HtmlEditorProviders (3 folders, A: 2, M: 2)

- [ ] DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE
- [ ] DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Utilities
- [ ] DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Web

## 7) DDRMenu (4 folders, A: 2, M: 12)

- [ ] DNN Platform/Modules/DDRMenu
- [ ] DNN Platform/Modules/DDRMenu/Common
- [ ] DNN Platform/Modules/DDRMenu/Properties
- [ ] DNN Platform/Modules/DDRMenu/TemplateEngine

## 8) HTML (8 folders, A: 19, M: 7)

- [ ] DNN Platform/Modules/HTML
- [ ] DNN Platform/Modules/HTML/Components
- [ ] DNN Platform/Modules/HTML/Controllers
- [ ] DNN Platform/Modules/HTML/Controls
- [ ] DNN Platform/Modules/HTML/Models
- [ ] DNN Platform/Modules/HTML/Mvc
- [ ] DNN Platform/Modules/HTML/Views
- [ ] DNN Platform/Modules/HTML/js

## 9) Skins (6 folders, A: 17, M: 1)

- [ ] DNN Platform/Website/Portals/_default/Containers/_default/Views
- [ ] DNN Platform/Website/Portals/_default/Skins/_default/Views
- [ ] DNN Platform/Skins/Aperture
- [ ] DNN Platform/Skins/Aperture/Views
- [ ] DNN Platform/Skins/Aperture/Views/partials
- [ ] DNN Platform/Skins/Aperture/containers/Views

# Not distributed (not critical)

## 10) Samples (13 folders, A: 25, M: 7)

- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Mvc
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Mvc/Controllers
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Mvc/Controls
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Mvc/Views/Contact
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/App_LocalResources
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/Controls
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/Models
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/Properties
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/Services
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/Views
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Razor/js
- [ ] DNN Platform/Modules/Samples/Dnn.ContactList.Spa

# Can be moved to other PR (we can start only with Razor+)

## 11) SPA Modules Support (1 folders, A: 0, M: 4)

- [ ] DNN Platform/Library/UI/Modules/Html5

## 12) MVC Modules Support (5 folders, A: 0, M: 5)

- [ ] DNN Platform/DotNetNuke.Web.Mvc/Framework/ActionResults
- [ ] DNN Platform/DotNetNuke.Web.Mvc/Framework/Controllers
- [ ] DNN Platform/DotNetNuke.Web.Mvc/Framework/Modules
- [ ] DNN Platform/DotNetNuke.Web.Mvc/Properties
- [ ] DNN Platform/DotNetNuke.Web.Mvc/Routing

Commit:
restrore use of clientdependency.core.mvc

Commit:
add use of css variables

Commit:
add support for report-to

Commit:
update web.config for mvc

Commit:
port all what is needed to run from dnn9 version

Commit:
first part of the html module : view control

sachatrauwaen committed:
Merge pull request #4 from sachatrauwaen/mvcpl2

Mvcpl2

Commit:
use dnn10 default.css

Commit:
make edit in  mvc work again

sachatrauwaen committed:
Merge pull request #5 from sachatrauwaen/html-module

Html module

donker committed:
Merge branch 'develop' into MVCPL

Commit:
fix html compile errors

donker committed:
Merge pull request #6 from sachatrauwaen/fix-html-module

fix html compile errors

Commit:
fix build of install package
- add missing skin files and skin build
- new mvc rewriter project (for install dependencies of dlls)
- web.config for mvcurlrewriter and mvc clientdependency
- CSP reporting url validatation fix

donker committed:
Merge pull request #7 from sachatrauwaen/install-mvc

fix build of install packages

Commit:
enhance ModuleViewContollerBase

sachatrauwaen committed:
Merge pull request #9 from sachatrauwaen/module

enhance ModuleViewContollerBase

donker committed:
Trying to fix skin packaging for MVC pipeline

donker committed:
Fix and task to test packages

donker committed:
Merge pull request #10 from donker/2025-04-15-skinpackage

Packaging for skin

Commit:
razor containers for the skin

Commit:
Merge branch 'MVCPL' into containers

donker committed:
Merge pull request #11 from sachatrauwaen/containers

Containers

sachatrauwaen committed:
fix double class

sachatrauwaen committed:
add support for mvc modules

sachatrauwaen committed:
a step farter in mvc module support

sachatrauwaen committed:
Merge branch 'MVCPL' into mvc-module

sachatrauwaen committed:
fix key same as mvc modules

sachatrauwaen committed:
make existing mvc modules work on the mvc pipeline
tested on https://github.com/dnnsoftware/Dnn.Platform.Samples.Mvc/tree/master/src/Dnn.ContactList.Mvc

donker committed:
Merge pull request #13 from sachatrauwaen/mvc-module

Mvc module

donker committed:
Merge branch 'develop' into MVCPL

# Conflicts:
#	DNN Platform/Website/release.config

donker committed:
Merge pull request #14 from donker/MVCPL

Merge latest DNN 10 changes

sachatrauwaen committed:
Make MyWork control mvc

sachatrauwaen committed:
make NewDDRMenu compatible with ddrmenu razor menu templates and use old INodeManipulator for compatibility

sachatrauwaen committed:
remove unused files

sachatrauwaen committed:
fix buit to include web.config in skin folder
remove mvcmenu, its not needed anymore
remove not used web.config becaude added to install package and not included in project

sachatrauwaen committed:
fix build of skin

sachatrauwaen committed:
fix skin build, issues when extension not provided

donker committed:
Merge pull request #15 from sachatrauwaen/html-mywork

Make MyWork control mvc

donker committed:
Merge pull request #16 from sachatrauwaen/ddrmenu

Ddrmenu

donker committed:
Merge branch 'release/10.0.1' into MVCPL

# Conflicts:
#	DNN Platform/Website/release.config

sachatrauwaen committed:
Add use of route with area for mvc pipeline modules

donker committed:
Merge pull request #20 from sachatrauwaen/area

Add use of route with area for mvc pipeline modules

donker committed:
Prevent any meddling of the module foldername for the purpose of area labeling

donker committed:
Merge pull request #22 from donker/routeareafixalt

Prevent any meddling of the module foldername for the purpose of area

sachatrauwaen committed:
Create the ModuleContext in the Initialize of the controller to give the possibility to use it in Initaliz overloads (like in webforms)

sachatrauwaen committed:
Merge branch 'MVCPL' into mvc-module-init

sachatrauwaen committed:
add page header and portal header to the layout

donker committed:
Merge pull request #24 from sachatrauwaen/mvc-module-init

Mvc module init

donker committed:
Merge pull request #25 from sachatrauwaen/fix-pageheader

add page header and portal header to the layout

donker committed:
Merge branch 'release/10.0.1' into tmpmerge

donker committed:
Merge pull request #32 from donker/tmpmerge

Merge latest DNN changes into this branch

donker committed:
Merge branch 'develop' into MVCPL

donker committed:
Merge branch 'develop' into tmpmerge

donker committed:
Fix method call

donker committed:
Fix build error

donker committed:
Fix for postback in MVC modules

donker committed:
Merge pull request #34 from donker/tmpmerge

Merge latest DNN dev commits and fix for MVC modules

donker committed:
Merge branch 'develop' into MVCPL

donker committed:
Merge branch 'develop' into tmp

donker committed:
Merge pull request #35 from donker/tmp

Merge latest development branch from core project

donker committed:
Code fixes for outdated code

donker committed:
Rolling back earlier change

donker committed:
Solution to get AF token to work on MVC pipeline

donker committed:
Merge pull request #36 from donker/aftoken

Solution to get AF token to work on MVC pipeline

donker committed:
Merge branch 'develop' into MVCPL

donker committed:
Merge branch 'develop' into MVCPL

# Conflicts:
#	DNN Platform/DotNetNuke.Web.Client/packages.config

sachatrauwaen committed:
fix typo

sachatrauwaen committed:
performance improvement

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
Module Controls : new way to make modules compatible with mvc pipeline

sachatrauwaen committed:
restructure ModuleControls to fit more the the ViewComponent patern

sachatrauwaen committed:
fix logo styling

sachatrauwaen committed:
fix html module

sachatrauwaen committed:
fix empty pane rendering

sachatrauwaen committed:
fix user skin object

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
rename test to demo
add readme

sachatrauwaen committed:
performance optimization for menu

sachatrauwaen committed:
add cursor ai ignore

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline' into ModuleControl

# Conflicts:
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Containers/SkinHelpers.Content.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/ModelFactories/PageModelFactory.cs
#	DNN Platform/Modules/HTML/Mvc/EditHTMLControl.cs

sachatrauwaen committed:
fix unit tests

sachatrauwaen committed:
add MvcControlClass to UI and db

sachatrauwaen committed:
ad mvcmodelclass to db for html module

sachatrauwaen committed:
add documentation for CSP

sachatrauwaen committed:
update mvcmoduleclass in manifest and remove sql migration script

sachatrauwaen committed:
fix folder and leanup

sachatrauwaen committed:
refactor class names

sachatrauwaen committed:
Cleanup

sachatrauwaen committed:
fix some names in the readme

sachatrauwaen committed:
Cleanup and ViewData to RazorModuleControl

sachatrauwaen committed:
remove RateLimit, it's for later

sachatrauwaen committed:
cleanup, namespace fix, better error error handling, fix viewdata

sachatrauwaen committed:
CLEANUP NAMESPACES

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
add ModulePartial helper

sachatrauwaen committed:
clean message when module control does not support mvc pipeline

sachatrauwaen committed:
show error when module dous not support mvc pl for page editors

sachatrauwaen committed:
add UI for Pipeline settings in Site and Page Settings

sachatrauwaen committed:
modify url rewriter to route to mvc controller depending on the new portal and page settings

sachatrauwaen committed:
MVC PL Popup Editing issue

sachatrauwaen committed:
fix module actions for modules in the mvc pipeline

sachatrauwaen committed:
add infrastructure for adding Page and module messages
PageService for Managing PageMessages and other page info like title, description, keywords, and other head tags

sachatrauwaen committed:
add missing page messages when error in query string and when message in http context

mitchelsellers commented:
@bdukes @donker I believe we discussed that this is just a reset PR and initial review, as it is into their branch correct?  High level review only?

valadas commented:
Awesome, I'll try to get my review in this weekend...

mitchelsellers commented on Build/Tasks/BuildPackages.cs:
```diff
@@ -0,0 +1,36 @@
+    [IsDependentOn(typeof(CreateUpgrade))]
+    [IsDependentOn(typeof(CreateNugetPackages))]
+    [IsDependentOn(typeof(GeneratePackagesChecksums))]
+    public sealed class BuildPackages : FrostingTask<Context>
```
What is the purpose/goal of this task addition?  I'm concerned that this will be deleting a source controlled file.

mitchelsellers commented on DNN Platform/DotNetNuke.ContentSecurityPolicy/BaseCspContributor.cs:
```diff
@@ -0,0 +1,33 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.ContentSecurityPolicy
```
Is this the same as the pending CSP PR into `develop`?

sachatrauwaen replied:
Yes it's the same. It need to be removed from here after the pr in develop is merged, so we can remerg it in this branch.
Or some other sujestion how to manage this ?

mitchelsellers commented on DNN Platform/DotNetNuke.Web.Client/Providers/DnnStandardRenderer.cs:
```diff
@@ -0,0 +1,32 @@
+        {
+            get
+            {
+                var settingsVersion = this.dnnSettingsHelper.AreCompositeFilesEnabled();
```
Given the changes to CDF (In pending state now) should this be removed?

mitchelsellers commented on DNN Platform/DotNetNuke.Web.Client/MvcClientResourceManager.cs:
```diff
@@ -0,0 +1,588 @@
+        private static readonly ReaderWriterLockSlim LockFileExistsCache = new ReaderWriterLockSlim();
+
+        /// <summary>Adds the necessary configuration to website root <c>web.config</c> to use the Client Dependency component.</summary>
+        public static void AddConfiguration()
```
Why is this beind done as a Web.Client item, shouldn't this be handled by the existing functionality?  I'm concerned this is going to introduce a duplicate or race condition

sachatrauwaen committed:
fix test of DnnController

sachatrauwaen committed:
Fix tests on DnnController.cs

sachatrauwaen committed:
Fix blank lines in DnnController.cs

sachatrauwaen committed:
remove blank linesDnnController.cs

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleSettingsViewController.cs:
```diff
@@ -0,0 +1,262 @@
+            }
+        }
+
+        private void CheckPermissions()
```
Same concern here as my previous comment on the similar code in ModuleSettingsController.cs

Also what is the difference between ModuleSettingscontroller and MOduleSettingsViewController.cs?

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleViewControllerBase.cs:
```diff
@@ -0,0 +1,299 @@
+            {
+                return this.ModuleContext.TabId;
+            }
+        }
```
If we expose ModuleContext, do we need to also expose those properties, could we make the API surface simper by limiting the redundant properties?

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleViewControllerBase.cs:
```diff
@@ -0,0 +1,299 @@
+            {
+                return this.PortalSettings.PortalAlias;
+            }
+        }
```
Same for PortalSetings which is already exposed, do we need those properties also exposed?

valadas reviewed (commented):
Ok, I found a bit of time and reviewed about the first 1/4 of this PR.

I think we should bring this PR to the minimal viable MVC working state (even if incomplete) just with the minimum to load up an mvc theme and module.

MVC Pipeline and CSP are 2 big ticket items and bringing both in the same PR makes putthing this in a reviewable state hard IMO. The goal of this PR was to have a baseline so that we can review smaller additions in an easier way.

Other than that, it does not have to be in this PR but I would like that we add our stylecop rules to the new projects to enforce our coding standards on this huge amount of new code.

I'll try to review some more the next coming days.

Thanks @sachatrauwaen for this huge contribution!

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Security/Controllers/ModulePermissionsGridController.cs:
```diff
@@ -0,0 +1,375 @@
+            }
+        }
+
+        public ActionResult Index(int tabId, int moduleId, bool inheritViewPermissionsFromTab)
```
What protects this endpoint?

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/dnn.json:
Can you explain the dnn.json files?

valadas commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Startup.cs:
```diff
@@ -0,0 +1,35 @@
+            services.AddTransient<IPageModelFactory, PageModelFactory>();
+            services.AddTransient<ISkinModelFactory, SkinModelFactory>();
+            services.AddTransient<IPaneModelFactory, PaneModelFactory>();
+            services.AddTransient<IContainerModelFactory, ContainerModelFactory>();
```
Is transient the right lifecycle for all of these

valadas commented on DNN Platform/DotNetNuke.Web.MvcWebsite/Controllers/ModuleActionsController.cs:
```diff
@@ -0,0 +1,53 @@
+    {
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public ActionResult Delete(ModuleActionsDeleteModel model)
```
What protects this endpoint?

valadas commented:
> > MVC Pipeline and CSP are 2 big ticket items and bringing both in the same PR makes putthing this in a reviewable state hard IMO. The goal of this PR was to have a baseline so that we can review smaller additions in an easier way.
> 
> There an separate PR for CSP on the develop branch.
> 
> But removing it from this PR needs to remove a lot of code. The idee is to replace the CSP project of this PR with the PR on develop when the PR is merged in develop.
> 
> Actually we heave a lot of sync issues between the differnt PR's. I don'tknow the best way. So your advice is welcome.

So if that is the case, we need to review/agree on the CSP one first before spending too much time reviewing this one...

bdukes commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Extensions/StartupExtensions.cs:
```diff
@@ -0,0 +1,36 @@
+                            type is { IsClass: true, IsAbstract: false });
+            foreach (var controller in mvcControllerTypes)
+            {
+                services.TryAddTransient(controller);
```
We're using scoped for web API controllers, but transient for MVC controllers:
https://github.com/dnnsoftware/Dnn.Platform/blob/3ce24744492a57f2802db7e7b5f338a77ecc0604/DNN%20Platform/DotNetNuke.Web/Api/Extensions/StartupExtensions.cs#L33
https://github.com/dnnsoftware/Dnn.Platform/blob/3ce24744492a57f2802db7e7b5f338a77ecc0604/DNN%20Platform/DotNetNuke.Web.Mvc/Extensions/StartupExtensions.cs#L33

Scoped it probably the right way

sachatrauwaen committed:
mvc pipeline doc

sachatrauwaen committed:
doc

sachatrauwaen committed:
modified proposition

sachatrauwaen committed:
modified proposition

sachatrauwaen committed:
fix layout

sachatrauwaen committed:
Revert "fix layout"

This reverts commit e07308a0baf1064bee15a54d16b19d54cc281cee.

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

# Conflicts:
#	mvc-pipeline.md

sachatrauwaen committed:
fix layout

sachatrauwaen committed:
fix layout

sachatrauwaen committed:
fix text

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
fix text

sachatrauwaen committed:
Fix typos in mvc-pipeline.md

sachatrauwaen committed:
blog post

sachatrauwaen committed:
rremove docs from this branch

sachatrauwaen committed:
Remove module settings, because actually out of scope of MVC pipeline and is actually not working

sachatrauwaen committed:
Remove CSP from mvc pipeline (it will be integrated later)
Remove module settings, because actually out of scope

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
remove CSP from html module and module settings for mvc pipeline

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/Website/release.config

donker committed:
First sweep of repairs after merging of latest dev branch

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' into ModuleControl

# Conflicts:
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Containers/SkinHelpers.Content.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleSettingsController.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleSettingsViewController.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Framework/JavascriptLibraries/MvcJavaScript.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Modules/ModuleHelpers.cs
#	DNN Platform/DotNetNuke.Web.MvcWebsite/Controls/ModuleActionsControl.cs
#	DNN Platform/Library/Entities/Modules/ControlInfo.cs
#	DNN Platform/Library/Services/Upgrade/Upgrade.cs
#	DNN Platform/Modules/HTML/Controllers/DNN_HTMLController.cs
#	DNN Platform/Modules/HTML/Controllers/MyWorkViewController.cs
#	DNN Platform/Modules/HTML/Controls/EditHTMLControl.cs

sachatrauwaen committed:
Merge branch 'ModuleControl' into mvcpl-selection

sachatrauwaen committed:
upgrade to new cdf

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' into mvcpl-popups

# Conflicts:
#	DNN Platform/DotNetNuke.Web.Client/MvcClientResourceManager.cs
#	DNN Platform/DotNetNuke.Web.MvcWebsite/Controllers/DefaultController.cs

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' into mvcpl-moduleactions

# Conflicts:
#	DNN Platform/DotNetNuke.Web.MvcPipeline/HtmlHelpers.cs
#	DNN Platform/DotNetNuke.Web.MvcWebsite/Controls/ModuleActionsControl.cs

sachatrauwaen committed:
upgrade to new cdf

sachatrauwaen committed:
add use of pageService

sachatrauwaen committed:
remove cliendependecy.mvc
update to use new cdf

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

donker committed:
Repairs for latest merge of develop

donker committed:
Remove non used dll from manifest

donker committed:
Fix to web.config

donker committed:
Remove more references to old clientdependency dll

donker committed:
Remove build task that slipped in

sachatrauwaen committed:
Replace IReasourcable with IPageContrtibutor to use new cdf

sachatrauwaen committed:
Merge branch 'mvcpl-messages' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/DotNetNuke.Abstractions/Pages/IPageService.cs
#	DNN Platform/DotNetNuke.Abstractions/Pages/PagePriority.cs
#	DNN Platform/DotNetNuke.Abstractions/Pages/readme.md
#	DNN Platform/DotNetNuke.Web.MvcPipeline/ModelFactories/SkinModelFactory.cs
#	DNN Platform/DotNetNuke.Web.MvcPipeline/Models/PageModel.cs
#	DNN Platform/Library/Services/Pages/PageExtensions.cs
#	DNN Platform/Library/Services/Pages/PageService.cs
#	DNN Platform/Website/Default.aspx.cs
#	DNN Platform/Website/Views/Default/Layout.cshtml

sachatrauwaen committed:
perf optimisation

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
cleanup and simplify due to new cdf

sachatrauwaen committed:
Cleanup
Remove that is actually not used.

sachatrauwaen committed:
finish skinobject
remove legacy skinobjects

sachatrauwaen committed:
test and fix some skinhelpers

sachatrauwaen committed:
fix typo

sachatrauwaen committed:
fix typo

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
refactor DI usage in html helpers

sachatrauwaen committed:
add html helper to CKE provider
refactor html to use CKE with settings of webforms

sachatrauwaen committed:
refactor use of DI in html helpers

sachatrauwaen committed:
razor module development guide for the mvc pipeline

sachatrauwaen committed:
enhance doc

sachatrauwaen committed:
enhance doc

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
fix mvc of html module

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

# Conflicts:
#	Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/DTO/SiteSettings/UpdateOtherSettingsRequest.cs

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

sachatrauwaen committed:
add razor module sample
update mvc sample
small fixes for spa module sample

sachatrauwaen committed:
enhance and fixes for razor sample

sachatrauwaen committed:
cleanup and refactor small stuff

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/DotNetNuke.Web.Client/packages.config
#	DNN Platform/DotNetNuke.Web.Mvc/Framework/Controllers/DnnController.cs
#	DNN Platform/Library/Entities/Urls/AdvancedUrlRewriter.cs
#	DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/packages.config
#	Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/Converters.cs

donker committed:
Fix build issues

sachatrauwaen committed:
add style check to mvcwebsite project

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
remove un neaded changes

donker committed:
Fix Razor module build script

donker committed:
More fixes to Razor module packaging

sachatrauwaen committed:
add stypecop to project and styles all files

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
apply stylecop

sachatrauwaen committed:
remove depreciations

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

# Conflicts:
#	Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/PagesControllerImpl.cs

thienvc commented:
This is fantastic, I've been waiting for this for a long time. It feels like DNN is experiencing a strong resurgence. I hope the core version is finalized soon so I can experience it and gradually migrate the webform modules.

sachatrauwaen committed:
remove MvcWebsite project
move MvcPipeline outside of Support Project
rename querystring  mvcpage= yes/no to pipeline = mvc/webforms

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
Merge branch 'development' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Web/EditorControl.cs
#	Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/Converters.cs

donker committed:
Fixes

sachatrauwaen committed:
fixes paths when dnn install in iis application folder

sachatrauwaen committed:
doc file for skin transformation

sachatrauwaen committed:
update doc

sachatrauwaen committed:
update doc

sachatrauwaen committed:
add general migration guide

sachatrauwaen committed:
Add MVC support to DDRMenu

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
refactor ddrmenu with support for mvc pipeline and remove newddrmenu

donker committed:
Fix to Spa module container control for MVC pipeline to cater for virtual roots

donker committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/dnnsoftware/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
add portal.css

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
add example in comment with internal routing

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
comment internal router example

sachatrauwaen committed:
fix delete module , module actions

sachatrauwaen committed:
add module caching support in the mvc pipeline

sachatrauwaen committed:
Update user guide with image and formatting changes

Added an image to the user guide and updated formatting.

sachatrauwaen committed:
enhance mvc rewriter to select automaticaly mvc route for edit module controls

sachatrauwaen committed:
fix translations
module actions for mvc sample
mvc support for edit of mvc sample
fix ajax controller calls when not loged in in mvc sample

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
Add Auto option to Pipeline site setting

sachatrauwaen committed:
Revise DNN MVC Pipeline User Guide

Updated user guide for MVC pipeline with corrections and clarifications.

sachatrauwaen committed:
Update documentation for Razor Module Development

sachatrauwaen committed:
add partialskin helper with parameters

sachatrauwaen committed:
fixes to guide

sachatrauwaen committed:
ensure the form is not shown

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

sachatrauwaen committed:
cleaup project file + apply style rules

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
remove unused usings

sachatrauwaen committed:
remove unused usings

sachatrauwaen committed:
remove all logic from models

sachatrauwaen committed:
fix: add area to route

sachatrauwaen committed:
cleanup

sachatrauwaen committed:
cleanup + fix build

sachatrauwaen committed:
Merge branch 'development' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/DotNetNuke.Web.Mvc/Framework/Controllers/DnnController.cs
#	DNN Platform/DotNetNuke.Web.Mvc/Framework/Modules/ResultCapturingActionInvoker.cs
#	DNN Platform/DotNetNuke.Web.Mvc/Routing/MvcRoutingManager.cs
#	DNN Platform/Library/Entities/Portals/PortalSettings.cs
#	DNN Platform/Library/Entities/Portals/PortalSettingsController.cs
#	DNN Platform/Library/Entities/Urls/AdvancedUrlRewriter.cs
#	DNN Platform/Library/UI/Modules/Html5/Html5HostControl.cs
#	DNN Platform/Library/UI/Modules/Html5/Html5ModuleControlFactory.cs
#	DNN Platform/Modules/DDRMenu/Common/DNNContext.cs
#	DNN Platform/Modules/DDRMenu/Localisation/Localiser.cs
#	DNN Platform/Modules/DDRMenu/MenuBase.cs
#	DNN Platform/Modules/DDRMenu/TemplateEngine/TemplateDefinition.cs
#	DNN Platform/Modules/HTML/Components/HtmlTextController.cs
#	DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/Web/EditorControl.cs
#	Dnn.AdminExperience/Dnn.PersonaBar.Extensions/admin/personaBar/Dnn.SiteSettings/App_LocalResources/SiteSettings.resx

sachatrauwaen committed:
fix routing issues - redifine more specific route for the mvcpipeline

sachatrauwaen committed:
fix csp separator and make it CLS compliant

sachatrauwaen committed:
add CSP to mvc pipeline

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

sachatrauwaen commented:
I’d like to check the status of this PR. There hasn’t been any review activity for several months—are there any issues or blockers? How can I help move things forward?

It might also be worth bringing this up during a TAG meeting ?

I completely understand that reviewing this PR is a significant effort. At the same time, on the otherhand inoticing quite a few major changes being made elsewhere, which also require substantial merge work on my side.

I believe it’s generally more efficient to process PRs in order (except for bug fixes, of course). Otherwise, each new PR tends to create additional merge overhead for those still pending.

thienvc commented:
> I’d like to check the status of this PR. There hasn’t been any review activity for several months—are there any issues or blockers? How can I help move things forward?
> 
> It might also be worth bringing this up during a TAG meeting ?
> 
> I completely understand that reviewing this PR is a significant effort. At the same time, on the otherhand inoticing quite a few major changes being made elsewhere, which also require substantial merge work on my side.
> 
> I believe it’s generally more efficient to process PRs in order (except for bug fixes, of course). Otherwise, each new PR tends to create additional merge overhead for those still pending.

I wish this PR was a custom package that could be installed and uninstalled like a plugin. 
If that were the case, I think it wouldn't affect the official version.

mitchelsellers commented on DNN Platform/DotNetNuke.ContentSecurityPolicy/CspExtensions.cs:
```diff
@@ -16,17 +16,46 @@ public static class CspExtensions
+
+            if (isAuthenticated)
+            {
+                csp.FrameSource.AddHost("https://dnndocs.com").AddHost("https://docs.dnncommunity.org");
```
I don't think this should be a global approval, this will result in some really funny security responses most likely

sachatrauwaen replied:
This urls are used by some helpurl of dnn core modules with are loaded in a iframe.
In place of add them globaly to csp policy, maybe they can be added only on the help page.
This this solve your concerns ? Or is it more about the url it self ?

valadas replied:
Do we actually load any resources from dnndocs? If so maybe we should rethink that as many implementations won't even be able to.

mitchelsellers replied:
My larger concern is that its included anywhere, rather than where only needed, so yes, if we could get it added into that module that woudl help.

sachatrauwaen replied:
it's moved to the help control

bdukes replied:
Should these be added dynamically, based on the help URL of the module? Will this still fail for 3rd party modules that have help content on other domains?

sachatrauwaen replied:
> Should these be added dynamically, based on the help URL of the module? Will this still fail for 3rd party modules that have help content on other domains?

I changed it that way. Dynamic add domain of helpurl.

mitchelsellers commented on DNN Platform/DotNetNuke.Web.Mvc/StandardTabAndModuleInfoProvider.cs:
```diff
@@ -99,6 +99,11 @@ private static int FindInt(HttpRequestBase requestBase, string key)
                 value = requestBase.Headers[key];
             }
 
+            if (requestBase.Form[key] != null)
```
I would like to review/discuss the possible risks wit htis change, by accpeting the value from the form in any situation , as I believe this might expose a different method of manipulation that could have unintended consequences.

sachatrauwaen replied:
because this, can be avoided and is based on convensions that are not common in mvc , i remove the change and avoid  unintended consequences

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Commons/PropertyHelper.cs:
```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+namespace DotNetNuke.Web.MvcPipeline.Commons
```
This entire implementation already exists within the solution at `DotNetNuke.Web.Mvc.Common`

Should this be a move?  Should the old file be marked as Obsolete?

sachatrauwaen replied:
removed and use the mvc version

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Commons/TypeHelper.cs:
```diff
@@ -0,0 +1,37 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Web.MvcPipeline.Commons
```
To match other naming conventions if needed this namespace/folder should be `Common` not `Commons`

sachatrauwaen replied:
i remove it and use mvc project version.

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/DnnPageController.cs:
```diff
@@ -0,0 +1,42 @@
+    /// <summary>
+    /// Base controller for DNN MVC page controllers, exposing common services and portal context.
+    /// </summary>
+    public abstract class DnnPageController : Controller, IMvcController
```
Similar question here for the fate of the existing IDNNController and related items, as I can see a lot of confusion without some level of guidance being provided at a programming perspective with different behaviors and purposes.  For example this ONLY provides PortalSettings the legacy one provides much more rich context (User, etc)

sachatrauwaen replied:
This one is indended to work in mvc pipeline, real mvc. IDNNController is for webforms , have a reference to a Page. Dous not use standard mvc. Use hacks in mvc to write the output to a string and generate module actions in a way that is not compatible with real mvc.

mitchelsellers replied:
@donker @valadas @bdukes @david-poindexter given the above comment from @sachatrauwaen do we need to think about the addition of an Obsolete or other attribute on the others?  I'm worried about multiple items that do similar things and those not being aware of what exactly this is, or this important difference

bdukes replied:
Are you suggesting that we mark the current MVC pattern as obsolete?

mitchelsellers replied:
Maybe?  I'm worried that with multiple ways of doing something, and/or without doing so, it is going to create even more confusion? Maybe it's too early for that?

bdukes replied:
Yeah, I think it's too early for that. We're just going to have to live with two MVC patterns side-by-side for a while.

sachatrauwaen replied:
In the mvc pipeline there is a way to run actual mvc modules as child controllers. But with some limitations. And when looking at the way actuel mvc modules are implemented is i think the only way that was possible in mvc 3. It's really bad hacking witch bring a lot of restructions to the real MS MVC. So opinion, is that depreciating the actual mvc modules is not a bad thing. Also because this patern can not work in dotnet core (because in use child controllers witch is not available in dotnet core). 

Depreciating the actual mvc modules, can also make this PR smalller.

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/ModuleViewControllerBase.cs:
```diff
@@ -0,0 +1,406 @@
+        /// Builds an edit URL for the current module instance.
+        /// </summary>
+        /// <returns>The edit URL.</returns>
+        public string EditUrl()
```
I echo @valadas comments on this duplication of exploding items that are already there, the long-term maintenance of this, as we have seen with other changes, is not exactly "fun".  I would prefer exposing the ModuleContext and leaving it at that personally

sachatrauwaen replied:
This a base controller is specifically for module controls implemented as a child controller. But is in reality not realy the recommended way because removed by MS in dotnet core. Maybe better to remove it ?

mitchelsellers replied:
I think in many ways, yes, it would be a "bigger lift" for somone converting an existing module, but at the same time, very easy document using ModuleContext

sachatrauwaen replied:
I remove the class, it is actually not used inside the mvc pipeline and the idea is to recommend the Razor+ pattern over Child Controller. So less code to maintain.

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Framework/JavascriptLibraries/MvcJavaScript.cs:
```diff
@@ -0,0 +1,338 @@
+                  .Register();
+
+            /*
+            if (Host.CdnEnabled && !string.IsNullOrEmpty(jsl.ObjectName))
```
This commented out code should be removed

sachatrauwaen replied:
this commended code, use DnnJsIncludeFallback with is actually not implemented in the mvc pipeline. Do you think this have some sense this days ? Is it needed to be implement in the mvc pipeline ?

bdukes replied:
This may have already been lost with @donker's CDF revamp.

sachatrauwaen committed:
remove code duplicated from mvc project

sachatrauwaen committed:
remove changes to StandardTabAndModuleInfoProvider

sachatrauwaen committed:
cleanup

sachatrauwaen commented:
> > I’d like to check the status of this PR. There hasn’t been any review activity for several months—are there any issues or blockers? How can I help move things forward?
> > It might also be worth bringing this up during a TAG meeting ?
> > I completely understand that reviewing this PR is a significant effort. At the same time, on the otherhand inoticing quite a few major changes being made elsewhere, which also require substantial merge work on my side.
> > I believe it’s generally more efficient to process PRs in order (except for bug fixes, of course). Otherwise, each new PR tends to create additional merge overhead for those still pending.
> 
> I wish this PR was a custom package that could be installed and uninstalled like a plugin. If that were the case, I think it wouldn't affect the official version.

We look at it, but not possible do the the use of lot of code that is not exposed for external project and need some UI changes in the core

sachatrauwaen committed:
add csp policy in the help control in place of globaly

bdukes commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Controllers/DnnPageController.cs:
```diff
@@ -0,0 +1,42 @@
+        /// <summary>
+        /// Gets the current portal settings.
+        /// </summary>
+        public PortalSettings PortalSettings
```
I would prefer that we only expose `IPortalSettings` (i.e. via the interface)

sachatrauwaen replied:
I have tryed, but endup in a nogo, because ActiveTab was not exposed thue IPortalSettings and found no way to get ActiveTab an oter way. Or do you know a way to get ActiveTab without PortalSettings ?

bdukes replied:
[`TabController.CurrentPage`](https://docs.dnncommunity.org/api/DotNetNuke.Entities.Tabs.TabController.html#DotNetNuke_Entities_Tabs_TabController_CurrentPage) is what I've been using. We want to have a better way to get that context, but it hasn't been fully designed yet.

sachatrauwaen replied:
I try againt to go for IPortalSettings but it need a lot of changes in cascade :
- HtmlTextContoller.FormatHtmlText
- HtmlTokenReplace
- TokenContext

It not realy cause breaking changes because IPortalSetings can accept Portalsettings parameters.
But is it a good idea to to all this changes inside this PR ?

Other issue is that IPortalSettings dous not implement IPropertyAccess in TokenReplace
this.AddPropertySource("portal", this.PortalSettings);
and also missing ActiveTab
this.AddPropertySource("tab", this.PortalSettings.ActiveTab);
So it need a more in detail restructuration of TokenReplace

To minimize the size of this PR i can use IPortalSettings in the controller and fallback to PortalSettings.Current when there is an issue with a class outside of the mvc pipeline project.

bdukes replied:
That sounds like a good compromise, thanks!

sachatrauwaen replied:
I apply it but a lot of PortalSettings.Current was needed.

sachatrauwaen committed:
remove unused class

sachatrauwaen committed:
restructure to use IPortalSettings , when i method require PortalSetting outsite of mvcpipelineproject PortalSettings.Current is used.
for PortalSettingsActiveTab that is not available in IPortalSettings, TabController.CurrentPage is used

mitchelsellers commented on Dnn.AdminExperience/Library/Dnn.PersonaBar.UI/admin/personaBar/Mvc/PersonaBarContainerController.cs:
```diff
@@ -0,0 +1,137 @@
+    public class PersonaBarContainerController : Controller
+    {
+#pragma warning disable CS0618 // Type or member is obsolete
+        private readonly IPersonaBarContainer personaBarContainer = Dnn.PersonaBar.Library.Containers.PersonaBarContainer.Instance;
```
This line and all others that are referencing Obsolete should be updated and the `#pragma` items go away

sachatrauwaen replied:
done

mitchelsellers commented on DNN Platform/Modules/Samples/Dnn.ContactList.Mvc/RouteConfig.cs:
```diff
@@ -10,7 +10,7 @@ public class RouteConfig : IMvcRouteMapper
         public void RegisterRoutes(IMapRoute mapRouteManager)
         {
-            mapRouteManager.MapRoute("ContactList", "ContactList", "{controller}/{action}", new[]
+            mapRouteManager.MapRoute("Dnn/ContactList", "ContactList", "{controller}/{action}", new[]
```
Why was this changed?  Was it broken before or does this create a needed breaking change for MVC modules?

sachatrauwaen replied:
1) This parameter is foldername witch is Dnn/ContactList (defined in the module manifest).
2) In the mvc pipeline old mvc modules run as child controllers and use mvc routing (witch is not realy the case in webforms). Child controllers can not use namespace routing but only area route value to differenciate 2 controllers with same name. So this first parameter is used as area name and it is required that it is equal to folder name of module definition.
3) As say before, the actual mvc modules contain big hack in the standard mvc. It can be an option, not to support actual mvc modules in mvc pipeline. I personnaly esitating.

bdukes replied:
are you saying that the sample module is currently broken without these changes? or did you adjust it for the MVC pipeline in a way that required these changes?

sachatrauwaen replied:
it is adjusted for the mvc pipeline and work also outside the mvc pipeline (in webforms)

sachatrauwaen committed:
removing Obsolete and #pragma

mitchelsellers commented on DNN Platform/DotNetNuke.Web.MvcPipeline/Containers/ContainerHelpers.Content.cs:
```diff
@@ -0,0 +1,97 @@
+            {
+                if (TabPermissionController.CanAdminPage())
+                {
+                    moduleDiv.InnerHtml += htmlHelper.ModuleErrorMessage(ex.Message, "Error loading module");
```
This should be localized for replacement

mitchelsellers replied:
Possibly look for others as well

sachatrauwaen committed:
fix getting portalsettings

donker committed:
Merge branch 'feature/mvc-pipeline' into feature/mvc-pipeline-old

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/Modules/HTML/web.config
#	DNN Platform/Providers/HtmlEditorProviders/DNNConnect.CKE/web.config

donker committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/dnnsoftware/Dnn.Platform into feature/mvc-pipeline-old

sachatrauwaen committed:
remove things that not exist anymore

sachatrauwaen committed:
Merge branch 'dnnsoftware:feature/mvc-pipeline-old' into feature/mvc-pipeline-old

donker committed:
Repair merge errors in advanced url rewriter

donker committed:
Move internalsvisible from separate file to main file

donker committed:
Fix merge error in DnnController

donker committed:
Merge branch 'develop' into feature/mvc-pipeline-old

sachatrauwaen committed:
start fixing mvc modules in mvc pipeline

sachatrauwaen committed:
Merge branch 'feature/mvc-pipeline-old' of https://github.com/sachatrauwaen/Dnn.Platform into feature/mvc-pipeline-old

# Conflicts:
#	DNN Platform/DotNetNuke.Web.Mvc/Framework/Controllers/DnnController.cs

sachatrauwaen committed:
start fixing mvc modules in mvc pipeline

dimarobert commented on DNN Platform/DotNetNuke.Web.MvcPipeline/ModuleControl/MvcModuleControl.cs:
```diff
@@ -0,0 +1,229 @@
+                    return string.Empty;
+                }
+
+                return segments[1];
```
I think this needs to be
```suggestion
                if (segments.Length == 3)
                {
                    return segments[1];
                }
                
                return segments[0];
```

formats for the ControlSrc are:
- `{ControllerName}/{ActionName}.mvc` - segments[0] needed here
- `{Namespace}/{ControllerName}/{ActionName}.mvc` - segments[1] needed here

Also, probably `RouteNamespace` below is valid only when `segments.Length == 3`.

dimarobert commented:
A took another look at whether there could be a way to support async MVC Modules with the new pipeline. The issue that we come to in [this thread](https://github.com/dnnsoftware/Dnn.Platform/pull/7193#issuecomment-4215415592), that prevents us from doing it boils down to the fact that the entire logic of the MVC Module execution in the new pipeline happens during the Razor View render phase of the page through a call to `HtmlHelper.Action(controllerName, actionName, routeValues)` and that the entire rendering phase in ASP.NET MVC is synchronous.

So my next thought was, instead of running this pipeline as it is running now:
1. Run **page controller phase** (`DefaultController.Page()` method) which builds the PageModel and results in a ViewResult.
2. Run the **page view rendering phase**, which uses the PageModel to output the HTML content of the page.
3. **For each** MVC Module on the page, during this phase, start yet another complete MVC sub-pipeline run (controller and view phase) using `HtmlHelper.Action()` which captures the result into a StringWriter and then appends it to the page output. (also generating a bunch of intermediary strings in memory in the process)

What if we could run it in the proper order, all the Controller phases first, Page controller then each Module controller, then similarly the View rendering phases?

To sanity check and see if it could maybe work from the point of view of what the APIs in ASP.NET MVC provide us, I worked through a possible implementation architecture / description with a bit of help from AI. The following proposal was the result of that after a couple of iterations, mostly output from the AI, with some of my observations as "[My note here: ...]" sections:

https://gist.github.com/dimarobert/ec0212c6de66085ba890187d1ca32368

I think that something like that could really work and it would even improve the order of execution a bit. Allowing even Module results that can redirect the request, coming in handy when implementing a [PRG pattern](https://en.wikipedia.org/wiki/Post/Redirect/Get).

donker committed:
Fix MVC module postback

sachatrauwaen committed:
fix isuue when usertabid is not initialised

sachatrauwaen commented:
> What if we could run it in the proper order, all the Controller phases first, Page controller then each Module controller, then similarly the View rendering phases?

Thanks for your comment. I have no easy response to your proposition. For sure there are some benefits. For now i give you only some general information from what i know.

- For sure async is the future but for async to be effetive it need to be used over whole line witch is difficult in .net framework and dnn. And not used the right way there can be locking problems in .net framework.
- One of the main goal of the mvcpipeline is to get shorter to .net core where ViewComponent sound like the best way to implement Module Controls.
- Actually the MVC module pattern of dnn is not the one best integrates in the MVC pipeline, because it trys to run a mvc controller as a child controller witch need routing with area witch was not intended to run like this in the webforms pipeline. And the new Razor pipeline fit most of the features of the actual dnn mvc module pattern.
- Initially a was thinking about a 3th alternative, where the controller phase will build the model of the page and also the models from all modules of the page and the View rendering then only render the layout and all modules models. Witch stay in a standard mvc concept of MVC (controller create the models, the views render the model).
- Actually we just hope this will be reviewed at some time by the core team to be integrated in a feature branch.  I don't think we will big restructuration before that.

dimarobert commented:
> Initially a was thinking about a 3th alternative, where the controller phase will build the model of the page and also the models from all modules of the page and the View rendering then only render the layout and all modules models. Witch stay in a standard mvc concept of MVC (controller create the models, the views render the model).

That is exactly what my proposed alternative above tries to do. It also gets rid of:
- the need for a child controller phase during the View phase.
- I do not think it needs routing with area (you just need the ControllerName and the ActionName from the ModuleControl definition to know what method to call)
- the need of rendering the modules to a bunch of StringWriters that are then appended to the page output in [MvcModuleControl.cs](https://github.com/dnnsoftware/Dnn.Platform/blob/feature/mvc-pipeline-old/DNN%20Platform/DotNetNuke.Web.MvcPipeline/ModuleControl/MvcModuleControl.cs#L223). (probably pretty costly as the output content of each module is manipulated at least 3 times before reaching the network).

Using the ViewComponent in net core or the current Razor pipeline for modules in DNN (basically module being able to run business logic only in the View phase) might work for modules that only render a shim HTML element and relies on a client-side js library (e.g. angular, react) to bootstrap the data, which would be retrieved through a secondary WebAPI call. But that can easily end up with a rendered page that is full of loading animations for each module, or maybe worse, a mostly empty page that starts filling up module by module. For modules that need to actually query and render data at page load (e.g. display a data table), having async would free up the request WorkerThread to handle other requests while the async I/O operation runs. It would be even more important for scenarios where MVC Views would be used for POSTing data back to the server, as that data would need to be processed, which is definitely slower.

> For sure async is the future but for async to be effetive it need to be used over whole line witch is difficult in .net framework and dnn. And not used the right way there can be locking problems in .net framework.

I am not sure this is true, async is safe in net framework and it is properly implemented in all three rendering technologies. MVC and WebApi are probably the safest, but WebForms is good as well. ASP.NET handles everything so that the execution can be async from top to bottom in the application code itself and you would never need to call `task.Result`, `task.Wait()`, or `task.GetAwaiter().GetResult()`, which are the APIs that on netfw could end up deadlocking on the SynchronizationContext. If you end up in a situation that you need to do that, then yes, it is very bad.
The same is equally true for net core if async methods are used from synchronous code and one of the three APIs above are used to wait their execution. There is indeed no SynchronizationContext, but you can as easily end up deadlocking if you do that.

sachatrauwaen committed:
cleanup
